server/fleet/datastore.go

```diff
@@ -1230,23 +1230,35 @@ type Datastore interface {
 	SetOrUpdateMDMAppleSetupAssistant(ctx context.Context, asst *MDMAppleSetupAssistant) (*MDMAppleSetupAssistant, error)
 	// Get the MDM Apple Setup Assistant for the provided team or no team.
 	GetMDMAppleSetupAssistant(ctx context.Context, teamID *uint) (*MDMAppleSetupAssistant, error)
+	// Get the MDM Apple Setup Assistant profile uuid and timestamp for the
+	// specified ABM token identified by organization name.
+	GetMDMAppleSetupAssistantProfileForABMToken(ctx context.Context, teamID *uint, abmTokenOrgName string) (string, time.Time, error)
```

Comment:
I set all those Datastore methods to expect the ABM token's org name, but maybe we'll end up having the token's id after all... Oh well, we'll still have the org name. It's not totally clear what the caller will have available at this point.